### PR TITLE
fix: Parse only openapi files

### PR DESCRIPTION
**Pull request by PelegLi**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

OpenAPI runner loads and parses many YML files unrelated to openapi (most YMLs aren't). Parsing many irrelevant files is time consuming, so in order to decrease the number of irrelevant files parsed, a soft pre-validation on the raw text file is added before parsing, excluding YMLs clearly unrelated to openapi.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: checkov/openapi/runner.py [hunk 1]
```diff
@@ -46,7 +46,14 @@ def parse_format(
         func: _ParseFormatJsonCallable | _ParseFormatYamlCallable,
     ) -> tuple[dict[str, Any] | list[dict[str, Any]], list[tuple[int, str]]] | None:
         try:
-            parsed_file = func(self, f)
+            if f.endswith(".json"):
+                parsed_file = func(self, f)
```
Comment (gruebel) [inline checkov/openapi/runner.py:50]: why not also do it for JSON files?

**Review by gruebel — COMMENTED** [review]
nice, always good to have a bit of pre evaluation.

**Review by rotemavni — COMMENTED** [review]
@PelegLi Can you create a test for the runner to scan a dir with multiple files, some for OpenAPI and some not, and validate the check results are as expected (as they were before the change)?

File: checkov/openapi/runner.py (hunk 1 above)
Comment (PelegLi) [reply checkov/openapi/runner.py:50]: 1. The main pain for openapi resides in the YAMLs, I want to check first if it improves performance and doesn't raise any other issues.
2. YAML runner allows passing the file content to the parser, so we don't need to open and load the file twice. JSON runner currently does not allow passing the content, and modifying it requires a deeper and riskier change.

Comment (gruebel) [reply checkov/openapi/runner.py:50]: sounds good 🙂

**Comment by PelegLi** [comment]
> @PelegLi Can you create a test for the runner to scan a dir with multiple files, some for OpenAPI and some not, and validate the check results are as expected (as they were before the change)?

Done

**Review by gruebel — APPROVED** [review]
☕

**Review by ayajbara — APPROVED** [review]

**Pull request merged**